### PR TITLE
White screen research plan

### PR DESCRIPTION
**Pull request by mekkyz**
Fixes https://github.com/ComPlat/chemotion_ELN/issues/1382

- Added UI error message to appear when backend fetching fails
- Modified UI (Added Edit button to be part of the research plan instead of own line, and other small things)

Before:

image

After:

image

### CODE REVIEW

**Comment by endevor-music** [comment]
> > Could you please provide images of your UI changes?
> 
> @endevor-music
> 
> I just updated the description:
> 
> The UI changes briefly:
> 
> 1. Edit button is integrated now in the research plan itself
> 2. Spacing between the close button and the private note
> 3. Spacing between elements
> 4. Outlining elements in view mode
> 5. Button sizes in edit and view mode

Amazing, thank you! I'll review this tomorrow.

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldReaction.js [hunk 1]
```diff
@@ -23,85 +27,221 @@ const hasAuth = (id) => {
...
+
+    let content;
+    if (reaction?.id) {
+      content = this.renderReaction(reaction);
+    } else if (wasReactionSet) {
+      content = (
+        <div style={{ color: 'red', textAlign: 'left' }}>
+          <i className="fa fa-exclamation-triangle" aria-hidden="true" style={{ marginRight: '5px' }} />
+          <span style={{ fontWeight: 'bold' }}>Element not found!</span>
+        </div>
```
Comment (endevor-music) [inline app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldReaction.js:238]: This div block appears multiple times (lines 147-150 & lines 211-214), could it be refactored into its own function, something like `renderElementError()` ?

**Review by endevor-music — CHANGES_REQUESTED** [review]

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldReaction.js (hunk 1 above)
Comment (mekkyz) [reply app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldReaction.js]: Fixed by: [776bbe9](https://github.com/ComPlat/chemotion_ELN/pull/1452/commits/776bbe91e314ea56c75cff3c481fa22e0319d56c)

File: app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldSample.js [hunk 2]
```diff
@@ -37,95 +45,204 @@ class ResearchPlanDetailsFieldSample extends Component {
...
+
+    let content;
+    if (sample?.id) {
+      content = this.renderSample(sample);
+    } else if (wasSampleSet) {
+      content = (
+        <div style={{ color: 'red', textAlign: 'left' }}>
+          <i className="fa fa-exclamation-triangle" aria-hidden="true" style={{ marginRight: '5px' }} />
+          <span style={{ fontWeight: 'bold' }}>Element not found!</span>
+        </div>
```
[the remark this answers is not in the document]
Comment (mekkyz) [reply app/packs/src/apps/mydb/elements/details/researchPlans/researchPlanTab/ResearchPlanDetailsFieldSample.js]: Fixed by: [776bbe9](https://github.com/ComPlat/chemotion_ELN/pull/1452/commits/776bbe91e314ea56c75cff3c481fa22e0319d56c)

**Review by endevor-music — DISMISSED** [review]
@PiTrem - ok to merge.

**Review by PiTrem — APPROVED** [review]

**Pull request merged**